packages/docusaurus-theme-common/removeThemeInternalReexport.mjs

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import fs from 'fs-extra';
+
+// See comment in src/internal.ts
+// This script should be run by CI tests to remove:
+// export * from './index'
+
+const filePath = 'lib/internal.js';
+const lineToRemove = "export * from './index';\n";
+
+if (!(await fs.pathExists(filePath))) {
+  throw new Error(`internal entrypoint file not found at ${filePath}`);
+}
+
+const fileContent = await fs.readFile(filePath, 'utf8');
+
+const fileContentUpdated = fileContent.replaceAll(lineToRemove, '');
+
+// Ensure the script correctly removes the re-export
+if (fileContent === fileContentUpdated) {
+  throw new Error(
+    'Unexpected: internal re-export has not been replaced.\nMake sure this script works, and is only run once.',
+  );
+}
+
+await fs.writeFile(filePath, fileContentUpdated);
```

Comment:
this idea is a bit convoluted but it looks to work fine 🤪 

The error message is not very visible though 😓 

![Uploading CleanShot 2022-06-23 at 22.00.05@2x.png…]()

Reply:
Lol, this looks like pure hack... Since we are to automate the marking of ejection safety by analyzing dependency on `/internal` APIs, we can simply reuse the same analyzer. I'll investigate that later.

Reply:
Yes 😅 maybe we'll remove this in the future once we have more public APIs

But this is not exactly the same problem: If a component is unsafe, it should still preferably import from the public endpoint instead of the internal re-export.